plugins/events-correlation-engine/src/main/java/org/opensearch/plugin/correlation/core/index/codec/correlation950/CorrelationCodec.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.correlation.core.index.codec.correlation950;
+
+import org.apache.lucene.codecs.Codec;
+import org.apache.lucene.codecs.FilterCodec;
+import org.apache.lucene.codecs.KnnVectorsFormat;
+import org.opensearch.plugin.correlation.core.index.codec.CorrelationCodecVersion;
+
+/**
+ * Correlation Codec class
+ *
+ * @opensearch.internal
+ */
+public class CorrelationCodec extends FilterCodec {
+    private static final CorrelationCodecVersion VERSION = CorrelationCodecVersion.V_9_5_0;
```

Comment:
Should we extract out the codec version as part of the config or to a single owner, as its being used across enums and CorrelationCodec definitions.